nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -322,23 +322,42 @@ geometry_msgs::msg::PoseStamped RegulatedPurePursuitController::getLookAheadPoin
   // If the no pose is not far enough, take the last pose
   if (goal_pose_it == transformed_plan.poses.end()) {
     goal_pose_it = std::prev(transformed_plan.poses.end());
-  } else if (params_->use_interpolation && goal_pose_it != transformed_plan.poses.begin()) {
+  }
+  if (params_->use_interpolation && goal_pose_it != transformed_plan.poses.begin()) {
```

Comment:
Interpolation is regarding interpolating between path points for sparse paths / large vehicles for selecting the lookahead points at steady state. 

I think this approach to goal stuff is something separate. Mostly because people might want this feature you're adding, but not care about path tracking interpolation

Reply:
My reasoning is that if someone isn't turning on the interpolation then they (presumably) care about only having carrots that are poses from the path. Projecting off the end of the path would violate that as we are creating a new pose to track.

WRT the default behaviour - yes, I believe it should always be enabled. At the moment if you are asking for interpolated carrots, you are in effect asking for the planner to always produce a carrot that is exactly the lookahead distance from the robot. At the moment when approaching the goal this is not the case - the carrot gets closer and closer.
For an Ackermann steered vehicle where the `base_link` is on the rear axle as you approach the goal the carrot ends up behind the front steering wheels. This makes it very unstable, and small cross-track errors lead to big deviations in yaw. With very little path left this very often ends up with the robot reaching the goal with very poor alignment (on our platform anyway!).

I have seen the other work done here on slowing down on approach to goal, and that is something that we are interested in because we also have a large platform with limits on how quickly it can speed up and slow down. But this PR isn't meant to address any of those things - just tracking of the path as the robot approaches the goal. All the calcs about when to start slowing down are done with reference to the last pose in the path, so this PR does not change that.

Reply:
The interpolation is mostly for smooth motion purposes so that between iterations that the velocity curves are more continuous -- which is important for larger vehicles that might have pretty sparse paths (this was created by an ag-tech company). I don't think the rationale is really about being on the exact path with exact path points. 

> WRT the default behaviour - yes, I believe it should always be enabled ... I have seen the other work done here on slowing down on approach to goal, But this PR isn't meant to address any of those things 

Ah, do you think we could? For instance
- Do this forward projection to the lookahead distance
- But reduce the velocity in proportion to the distance remaining to the actual goal pose

If we had something like that, we could simply make it default then without a parameterization. In fact, isn't that the behavior with this enabled + `approachVelocityConstraint` (which is non-optional)? I believe the slowdown logic is already decoupled the lookahead distance

Reply:
> The interpolation is mostly for smooth motion purposes so that between iterations that the velocity curves are more continuous -- which is important for larger vehicles that might have pretty sparse paths (this was created by an ag-tech company). I don't think the rationale is really about being on the exact path with exact path points.

I was just guessing - I would never want to run without the interpolation on for exactly the reasons you state! ;) Even if the path wasn't sparse.

> If we had something like that, we could simply make it default then without a parameterization. In fact, isn't that the behavior with this enabled + approachVelocityConstraint (which is non-optional)? I believe the slowdown logic is already decoupled the lookahead distance

Yes, the slowdown is already decoupled - it uses the current pose and last pose in the path as parameters. The carrot is not involved in those calculations.

Reply:
Honestly, this might be a kick for me to disable the option for interpolation and just have it happen. That was a new feature so I didn't want to break users, but its had enough baking time now I think it can just be always included. But something to do after this PR is figured out!